### ISSUE TITLE
Add Linux platform detection to `twisted.python.runtime`

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5491]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5491|
|Type|enhancement|
|Created|2012-02-18 16:06:27Z|
|Branch|https://github.com/twisted/twisted/tree/islinux-5491|

With Linux 3 out and starting to see some use, various bits of code checking `sys.platform` against the literal string "linux2" are becoming outdated and will start making the wrong decision.

Rather than duplicate this logic throughout, we should have `Platform.isLinux()` to answer the question.

The first such fix was made for [#5489](https://github.com/twisted/twisted/issues/5489).  I expect that [#5478](https://github.com/twisted/twisted/issues/5478) may also benefit from being able to make such a check.

<details><summary>Searchable metadata</summary>

```
trac-id__5491 5491
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_islinux_5491 branches/islinux-5491
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1329581187000000 1329581187000000
changetime__1329784028000000 1329784028000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [33583]) Branching to 'islinux-5491'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst commented|
|-|-|

Just need to run buildbot on the branch and if it passes it's ready for review, I'll do that in the morning.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

For your convenience, [the build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/islinux-5491).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @itamarst|
|-|-|

I wonder if `test_isLinuxConsistency` should bother doing anything with `platform.getType()`.  The implementation of `isLinux` doesn't check it, so conceivably a Python runtime could tell us `sys.platform` is `"linux"` when `os.name` is, say, `"winnt"` - the test would fail.  Alternatively, maybe `isLinux` should bother checking for POSIX.

Address one way or another and then merge.  Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_6"></a>|@itamarst set status to `closed`|
|-|-|

(In [33587]) Merge islinux-5491: Add ability to check if platform is Linux.

Author: itamar
Review: exarkun
Fixes: [#5491](https://github.com/twisted/twisted/issues/5491)

Add t.p.runtime.platform.isLinux() method.
